docs/MO_DG/Deep_Learning_Model_Optimizer_DevGuide.md

```diff
@@ -118,3 +118,70 @@ For more information, refer to the [Converting a Kaldi Model](prepare_model/conv
 refer to the [Model Conversion Tutorials](prepare_model/convert_model/Convert_Model_Tutorials.md).
 - For more information about IR, see [Deep Learning Network Intermediate Representation and Operation Sets in OpenVINO™](IR_and_opsets.md).
+
+## MO Python API
+
+- Model Optimizer has Python API for model conversion, which is represented by convert_model() method in openvino.tools.mo namespace.
+  convert_model() has all the functionality available from command line tool plus the possibility of passing Python objects (models, extensions) to convert_model() directly from memory.
+  convert_model() returns openvino.runtime.Model object which can be compiled and infered or serialized to IR.
+
+```sh
+from openvino.tools.mo import convert_model
+
+ov_model = convert_model("resnet.onnx")
+```
+
+MO Python API allows conversion of PyTorch models. Converting to PyTorch requires providing "input_shape" or "example_input".
+```sh
+import torchvision
+
+model = torchvision.models.resnet50(pretrained=True)
+ov_model = convert_model(model, input_shape=[1,3,100,100])
+```
+
+For PyTorch conversion MO exports the model to ONNX as intermediate step. ONNX opset version can be set by parameter "onnx_opset_version".
+```sh
+import torchvision
+import torch
+from openvino.tools.mo import convert_model
+
+model = torchvision.models.resnet50(pretrained=True)
+ov_model = convert_model(model, example_input=torch.zeros(1, 3, 100, 100), onnx_opset_version=13)
+```
+
+- Following types are supported as input model for convert_model():
+
+| PyTorch                                                          |                                                                 TF2 / Keras                                                                 |                                   TF                                   |
+|:-----------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------:|:----------------------------------------------------------------------:|
+| torch.nn.Module torch.jit.ScriptModule torch.jit.ScriptFunction  | tf.keras.Model tf.keras.layers.Layer tf.function tf.Module tf.train.checkpoint tf.python.training.tracking.base.Trackable(with limitations) | tf.compat.v1.GraphDef tf.compat.v1.wrap_function tf.compat.v1.session  |
+
+- convert_model() accepts all parameters available in MO command line tool. Parameters can be specified by Python classes or string analogs identically to command line tool.
+- Example 1:
+```sh
+from openvino.runtime import PartialShape, Layout
+
+ov_model = convert_model(model, input_shape=PartialShape([1,3,100,100]), mean_values=[127, 127, 127], layout=Layout("NCHW"))
+```
+- Example 2:
+```sh
+ov_model = convert_model(model, input_shape="[1,3,100,100]", mean_values="[127,127,127]", layout="NCHW")
```

Comment:
for me not enough info about `compress_to_fp16` and other arguments which have not values as command line parameters